### PR TITLE
Refactor api & src subdirectories to use config.json

### PR DESCRIPTION
**Pull request by williamtnguyen**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -81,4 +81,5 @@ build
 printingRPC/
 # Config files
 config.js
```
Comment (evanugarte) [inline .gitignore:83]: remove this line as we arent using it anymore

File: api/config/config.example.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  },
+  "googleCalendarIds": {
+    "PRIMARY_CAL": "primary",
+    "SCE_EVENTS_CAL": "XXXXXXXXXXXXXXXXXXXXXXXXXXX"
+  },
+  "ledSignIp": "192.168.4.1",
+  "GENERAL_API_URL": "http://localhost:8080/",
+  "LOGGING_API_URL": "http://localhost:8081/",
+  "MAILER_API_URL": "http://localhost:8081/"
+}
```
Comment (evanugarte) [inline api/config/config.example.json:19]: add blank line at bottom of this file

File: src/APIFunctions/2DPrinting.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import axios from 'axios';
 import { PrintApiResponse } from './ApiResponses';
-import { RPC_API_URL } from '../config/config';
+const { RPC_API_URL } = require('../config/config.json');
```
Comment (evanugarte) [inline src/APIFunctions/2DPrinting.js:3]: for importing stuff in `src/` use the convention like the line above:

```suggestion
import { RPC_API_URL } from '../config/config.json';
```
please fix for all other files in `src`

**Comment by williamtnguyen** [comment]
This PR seems to not build whereas the other PR builds because the `config.json`'s in this PR are .gitignore'd

**Comment by evanugarte** [comment]
@williamtnguyen the issue here is that when travis builds, `config.json` doesn't exist.

There are two options to fix this:
1. Create a base64 encoded environment variable for each config file and decode it to both `api/config/config.json` and `src/config/config.json`.
2. Run `setup.py` in the `install` section in `.travis.yml` like:

```yml
install:
  python setup.py
  # rest of stuff here ...
```
Note that you should remove the `setup_status` line for each os, as it clones SCE-RPC but is no longer required for tests to pass.
image

**Comment by jerrylee17** [comment]
Rebase

File: api/config/config.example.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    "REDIRECT_URIS": ["https://developers.google.com/oauthplayground"],
+    "USER": "sce.sjsu@gmail.com",
+    "REFRESH_TOKEN": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+  },
+  "googleCalendarIds": {
+    "PRIMARY_CAL": "primary",
+    "SCE_EVENTS_CAL": "XXXXXXXXXXXXXXXXXXXXXXXXXXX"
+  },
+  "ledSignIp": "192.168.4.1",
+  "GENERAL_API_URL": "http://localhost:8080/",
```
Comment (evanugarte) [inline api/config/config.example.json:16]: remove the slashes at the end of each URL in this file

File: api/config/config.example.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  "LOGGING_API_URL": "http://localhost:8081/",
+  "MAILER_API_URL": "http://localhost:8081/"
```
Comment (evanugarte) [inline api/config/config.example.json:18]: this should be 8082

**Pull request merged**